src/lib/redirects-manager-context.tsx

```diff
@@ -0,0 +1,116 @@
+import * as React from 'react';
+import { v4 } from 'uuid';
+import { saveState, checkRepeatedRequests } from './utils';
+
+export interface RedirectsManagerContextInterface {
+    store: RedirectionsStore;
+    wildcard: boolean;
+    saving?: boolean;
+    lastSave?: Date;
+    lastModification?: Date;
+    filterBy?: string;
+    perPage?: number;
+    currentPage?: number;
+    imported?: boolean;
+}
+
+export interface RedirectionProps {
+    id: string;
+    request?: string;
+    destination?: string;
+    modificationDate?: any;
+    warningRequestDuplication?: boolean;
+}
+
+export interface RedirectionsStore extends Array<RedirectionProps> { }
+
+type Dispatch = (action: Action) => void;
+interface RedirectsManagerProviderProps { children: React.ReactNode; }
+
+const RedirectsManagerContext = React.createContext<RedirectsManagerContextInterface | undefined>(undefined);
+const RedirectsManagerDispatchContext = React.createContext<Dispatch | undefined>(undefined);
+
+interface UpdateServerStateProps {dispatch: Dispatch; state: RedirectsManagerContextInterface; }
+
+export const updateServerState = async ({dispatch, state}: UpdateServerStateProps) => {
+    dispatch({type: 'saving-state', value: true});
+    try {
+        const newState = await saveState(state);
+        dispatch({type: 'set', value: {...newState, lastSave: new Date(), saving: false}});
+    } catch (e) {
+        dispatch({type: 'saving-state', value: false});
+    }
+};
+
+type Action = {type: 'saving-state', value: boolean} |
```

Comment:
please group types / interface and order them to top.

Reply:
applied
Any upside on placing them on top?

Reply:
yes the code is more structured and not so fragmented